src/dex-ui-components/presets/SwapTokensForm/SwapTokensForm.tsx

```diff
@@ -207,64 +235,49 @@ for ${formValues.tokenToReceive.amount.toFixed(8)} ${formValues.tokenToReceive.s
   }
 
   function handleTokenToTradeAmountChanged(updatedToken: TokenState) {
-    const tokenToReceiveAmount = calculateTokenAmountToReceive({
-      tokenToTradeAmount: updatedToken.amount,
-      tokenToTradeLiquidity: updatedToken.poolLiquidity ?? 0,
-      tokenToReceiveLiquidity: formValues.tokenToReceive.poolLiquidity ?? 0,
-    });
-    const updatedTokenToReceive = {
-      ...formValues.tokenToReceive,
-      amount: tokenToReceiveAmount.toNumber(),
-      displayAmount: tokenToReceiveAmount.gt(0) ? String(tokenToReceiveAmount) : "",
-    };
-    updateTokenToReceiveAmounts(updatedTokenToReceive.amount, updatedTokenToReceive.displayAmount);
+    const tokenToTradeQty = updatedToken.amount;
+    if (tokenToTradeQty) {
+      fetchSwapPair(
+        updatedToken.tokenMeta.tokenId ?? "",
+        formValues.tokenToReceive.tokenMeta.tokenId ?? "",
+        updatedToken.amount === 0 ? 1 : updatedToken.amount
+      );
```

Comment:
` updatedToken.amount === 0 ? 1 : updatedToken.amount` this condition is due to the SC method limitation in which the Method does not work with `0` quantity send and we need to call the method to fetch the `Pair Id`, `PairInfo` and `Pool Liquidity`